socket.go

```diff
@@ -35,21 +38,30 @@ func newSocketServer(broker *Broker, auth *Auth, editable bool, baseURL string,
 }
 
 func (s *SocketServer) ServeHTTP(w http.ResponseWriter, r *http.Request) {
+	conn, err := upgrader.Upgrade(w, r, nil)
+	if err != nil {
+		echo(Log{"t": "socket_upgrade", "err": err.Error()})
+		return
+	}
+
 	session := anonymous
 	if s.auth != nil {
 		session = s.auth.identify(r)
 		if session == nil {
-			http.Error(w, http.StatusText(http.StatusUnauthorized), http.StatusUnauthorized)
+			// As per websocket spec, clients are not required to follow HTTP redirects. So we send a redirect message.
+			if msg, err := json.Marshal(OpsD{U: s.baseURL + "_auth/logout"}); err == nil {
```

Comment:
Does the client handle this message already?

Reply:
Yes - https://github.com/h2oai/wave/blob/42008b74a65520c3adc48c4f1f0bd2e22445b3e8/ui/src/ui.ts#L136.